apps/service_providers/llm_service/main.py

```diff
@@ -168,6 +172,15 @@ def transcribe_audio(self, audio: BytesIO) -> str:
         )
         return transcript.text
 
+    def attach_built_in_tools(self, built_in_tools: list[str], tools: list[BaseTool]):
```

Comment:
Instead of passing tools in here it's better to keep functions stateless where possible. So you should return the tool value and let the caller append it.